### DISCUSSION TITLE
Video orientation tools do not work correctly

### DISCUSSION THREAD

**Discussion by TrickPhysics** [smplayer-dev/smplayer#778] [General]
Hey all, 
I'm using:  Version: 23.6.0 (revision 10157) (64-bit) // Windows 10 22H2

Prior to my recent update, I had been using an older version of SMplayer (21.1.0 I believe), and I don't recall these issues existing in that old version, but regardless, here's the sitch.

1) When playing a video, it always opens in landscape.   Frequently, I am unable to use the Rotate, Flip, or Mirror options -- using them has no effect on the picture, it just keeps playing, *even though the Rotate menu shows the correct option when I go back*.

*Possibly useful info:  I am able to "fix" this by changing the video output driver to "direct3d"  I have also had luck at times using mplayer instead of mpv, but this doesn't always work; the direct3d switch seems to always make the controls work.
*Even after changing the rotation, opening the file again requires re-selecting the rotation.  Not sure if intended or not.

2) the older version of SMPlayer used to automatically rotate videos to portrait, and it did a very good job of getting it right the vast majority of the time.  I don;'t see this option anywhere, and it never does it by itself.  This is very useful when going back and forth between many video files quickly.  Should this option be somewhere?  

Important context:
These SAME videos, when played in VLC, Windows Media Player, and web browsers always (not completely verifiable) auto-rotate them successfully without user intervention, and in the case of VLC and Windows Media Player, I am able to manually rotate/flip with the player controls without any issues.

Previous references:  I found a couple forum posts on I think Stack Exchange that described the exact same problem with SMplayer, but they were back in 2020, with older versions and there were no posted successful answers.  I find it hard to believe that a bug like this has persisted for years, so that's when I came over here to ask around.

Anyway, I'm just at a loss here, and would love for anyone to give me suggestions or something to try.  SMplayer is by far my most preferred player, so I'm eager to get it working as expected again.

Thank you!

**Comment by Raza-0007** [comment]
>Anyway, I'm just at a loss here, and would love for anyone to give me suggestions or something to try. SMplayer is by far my most preferred player, so I'm eager to get it working as expected again.

@TrickPhysics, Very simple, go to

Options>Preferences>Performance>Decoding

and make sure hardware decoding is either set to "none" or "auto-copy".

SMPlayer video filters are applied in CPU and if GPU based hwdec is enabled, they will not work. Or you can have GPU based hwdec enabled but then copy back the decoded frame from GPU to CPU with "copy-back", then smplayer filters will work.

This holds true for all smplayer video filters like deinterlace, deblock, deband, flip, mirror, rotate etc.

**Comment by TrickPhysics** [comment]
Fantastic!  Thank you so much, that indeed fixed it! 

Appreciate your response and describing why the issue comes about.

Thanks a ton!

**Comment by TrickPhysics** [comment]
Closing as resolved.
